### PR TITLE
[tests-only] [full-ci] Refactoring unshare step definition code

### PR DESCRIPTION
**Pull request by S-Panta**
## Description
This PR refactors the step definition code relating to unshare scenerio.

## Related Issue
- Part of  <https://github.com/owncloud/ocis/issues/6325#issuecomment-1590750636>
## How Has This Been Tested?
- test environment:
- locally
- ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-2</strong>](https://drone.owncloud.com/owncloud/ocis/23569/59/1) failed. Further test are cancelled...

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -3491,6 +3491,25 @@ public function assertNoSharesOfUserAreInState(string $user, string $state):void
...
+	 * @When /^user "([^"]*)" unshares (?:folder|file|entity) "([^"]*)" shared to "([^"]*)"$/
+	 *
+	 * @param string $sharer
+	 * @param string $path
+	 * @param string $sharee
+	 *
+	 * @return void
+	 * @throws JsonException
+	 */
+	public function userUnsharesSharedTo(string $sharer, string $path, string $sharee): void {
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/Sharing.php:3512]:
```suggestion
	public function userUnsharesResourceSharedTo(string $sharer, string $path, string $sharee): void {
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -3354,58 +3354,51 @@ public function userDeletesFile(string $user, string $file):void {
...
 	 * @param string $user
-	 * @param string $deletedOrUnshared
-	 * @param string $entry
+	 * @param string $resource
 	 *
 	 * @return void
 	 * @throws Exception
 	 */
-	public function userHasDeletedFile(string $user, string $deletedOrUnshared, string $entry):void {
+	public function userHasDeletedFile(string $user, string $resource):void {
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/WebDav.php:3365]:
```suggestion
	public function userHasDeletedResource(string $user, string $resource):void {
```

File: tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature [hunk 3]
```diff
@@ -14,11 +14,11 @@ Feature: shares are received in the default folder for received shares
     And user "Alice" has created folder "FOLDER"
     When user "Alice" shares folder "/FOLDER" with user "Brian" using the sharing API
     And user "Brian" accepts share "/FOLDER" offered by user "Alice" using the sharing API
-    And user "Brian" unshares folder "Shares/FOLDER" using the WebDAV API
+    When user "Brian" declines share "/Shares/FOLDER" offered by user "Alice" using the sharing API
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature:17]:
```suggestion
    And user "Brian" declines share "/Shares/FOLDER" offered by user "Alice" using the sharing API
```

File: tests/acceptance/features/coreApiShareManagementBasicToShares/deleteShareFromShares.feature [hunk 4]
```diff
@@ -226,3 +226,13 @@ Feature: sharing
       | ocs_api_version | ocs_status_code | http_status_code |
       | 1               | 404             | 200              |
       | 2               | 404             | 404              |
+
+
+   Scenario: unshare a shared resources
+     Given using OCS API version "1"
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiShareManagementBasicToShares/deleteShareFromShares.feature:232]: also `use-ocs-api version 2`

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 5]
```diff
@@ -3338,7 +3338,7 @@ public function fileHasBeenDeleted(string $file, string $user):void {
 	}
 
 	/**
-	 * @When /^user "([^"]*)" (?:deletes|unshares) (?:file|folder) "([^"]*)" using the WebDAV API$/
+	 * @When user :user deletes file/folder :file using the WebDAV API
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/WebDav.php:3341]:
```suggestion
	 * @When user :user deletes file/folder :resource using the WebDAV API
```

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 2 above)
Comment (S-Panta) [reply tests/acceptance/features/bootstrap/WebDav.php:3365]: There are many step definition codes like this WebDav and Sharing files where similar refactoring can be done. 
Example: the word entity can be replaced in every code by 'resources', and the regex statement also looks redundant in the code.
@saw-jan @amrita-shrestha @grgprarup @PrajwolAmatya @SwikritiT

File: tests/acceptance/features/coreApiVersions/fileVersions.feature [hunk 6]
```diff
@@ -421,7 +421,7 @@ Feature: dav-versions
       | new         | Brian | /testshare |
 
 
-  Scenario: receiver tries to get file versions of unshared file from the sharer
+  Scenario: receiver tries to get file versions of file not shared by the sharer
```
[the remark this answers is not in the document]
Comment (S-Panta) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature:424]: How will this make the scenario statement clearer?

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 2 above)
Comment (saw-jan) [reply tests/acceptance/features/bootstrap/WebDav.php:3365]: IMO, those can be done in other refactor PR is they are not related to unshare and delete steps

File: tests/TestHelpers/SharingHelper.php [hunk 7]
```diff
@@ -158,7 +158,6 @@ public static function createShare(
 			$fd['expireDate'] = $expireDate;
 		}
 		$headers = ['OCS-APIREQUEST' => 'true'];
-
```
Comment (saw-jan) [inline tests/TestHelpers/SharingHelper.php:161]: It would be good for the reviewers (also for your PR) if you can omit these kind of changes

File: tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature [hunk 8]
```diff
@@ -14,11 +14,11 @@ Feature: shares are received in the default folder for received shares
...
     And user "Brian" shares folder "/Shares" with user "Alice" using the sharing API
     Then the OCS status code of responses on each endpoint should be "<ocs_status_code>" respectively
     And the HTTP status code of responses on each endpoint should be "<http_status_code>" respectively
     Examples:
-      | ocs_api_version | ocs_status_code | http_status_code   |
-      | 1               | 100, 100, 400   | 200, 200, 204, 200 |
-      | 2               | 200, 200, 400   | 200, 200, 204, 400 |
+      | ocs_api_version | ocs_status_code  | http_status_code   |
+      | 1               | 100, 100,100,400 | 200, 200, 200, 200 |
+      | 2               | 200, 200,200,400 | 200, 200, 200, 400 |
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature:24]:
```suggestion
      | 1               | 100, 100, 100, 400 | 200, 200, 200, 200 |
      | 2               | 200, 200, 200, 400 | 200, 200, 200, 400 |
```

File: tests/acceptance/features/coreApiVersions/fileVersions.feature (hunk 6 above)
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature:424]: `Scenario: share receiver tries to get file versions of file `

File: tests/TestHelpers/SharingHelper.php (hunk 7 above)
Comment (saw-jan) [reply tests/TestHelpers/SharingHelper.php:161]: I don't think so but it's fine for now.

File: tests/acceptance/features/coreApiVersions/fileVersions.feature (hunk 6 above)
[the remark this answers is not in the document]
Comment (S-Panta) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature:424]: Here, the sharer makes two files, shares only one of them and the sharee tries to see file versioning of the file not shared by the sharer.

[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature:424]: Ahh! then looks good :+1:

File: tests/acceptance/features/coreApiVersions/fileVersions.feature [hunk 9]
```diff
@@ -421,7 +421,7 @@ Feature: dav-versions
       | new         | Brian | /testshare |
 
 
-  Scenario: receiver tries to get file versions of unshared file from the sharer
+  Scenario: sharee tries to get file versions of file not shared by the sharer
```
Comment (amrita-shrestha) [inline tests/acceptance/features/coreApiVersions/fileVersions.feature:424]:
```suggestion
  Scenario: sharee tries to get file versions of another user's file
```

Comment (amrita-shrestha) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature:424]: another thing I don't understand is why we have to share the files step below. We can remove these steps
```
    And user "Alice" has uploaded file with content "textfile0" to "textfile0.txt"
    And user "Alice" has shared file "textfile0.txt" with user "Brian"
    And user "Brian" has accepted share "/textfile0.txt" offered by user "Alice"
```

File: tests/acceptance/features/coreApiVersions/fileVersions.feature (hunk 6 above)
[the remark this answers is not in the document]
Comment (amrita-shrestha) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature]: there is still no point to improve

File: tests/acceptance/features/coreApiVersions/fileVersions.feature (hunk 9 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature:424]: @saw-jan @SagarGi @SwikritiT  what you guys say!  do we need to add share steps in this scenario ?????

Comment (saw-jan) [reply tests/acceptance/features/coreApiVersions/fileVersions.feature:424]: IMO, should be fine there. This scenario tests the edge case that when textfile0 is shared with Brian, he cannot view the versions of another unshared file of Alice.

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

**Review by saw-jan — APPROVED** [review]
